sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -2037,16 +2058,53 @@ public void setResponsibleSurveillanceOfficer(Case caze) {
 		}
 	}
 
-	public void reassignTasks(Case caze) {
+	/**
+	 * Reassigns tasks related to `caze`. With `forceReassignment` beeing false, the function will only reassign
+	 * the tasks if the assignees lack jurisdiction on the case. When forced, all tasks will be reassigned.
+	 * 
+	 * @param caze
+	 *            the case which related tasks are reassigned.
+	 * @param forceReassignment
+	 *            force reassignment of case tasks.
+	 */
+	public void reassignTasksOfCase(Case caze, boolean forceReassignment) {
+		// for each task that is related to the case, the task assignee must match the jurisdiction of the case
+		// otherwise we will reassign the task
 		for (Task task : caze.getTasks()) {
 			if (task.getTaskStatus() != TaskStatus.PENDING) {
 				continue;
 			}
 
-			assignOfficerOrSupervisorToTask(caze, task);
+			User taskAsignee = task.getAssigneeUser();
+			boolean mismatch;
+
+			if (taskAsignee == null) {
+				// no one is assigned so we skip detailed checks and go directly reassignment.
+				mismatch = true;
+			} else {
+				boolean regionMismatch = !taskAsignee.getRegion().getUuid().equals(caze.getRegion().getUuid());
+				boolean districtMismatch = !taskAsignee.getDistrict().getUuid().equals(caze.getDistrict().getUuid());
+
+				boolean communityMismatch = taskAsignee.getCommunity() != null
```

Comment:
For community and health facility, the situation where 
* the user has a community/facility
* the case does not have a community/facility

also has to count as a mismatch (because it means the case is no longer associated with their jurisdiction)

Reply:
added :)